src/cli/subcommand/uninstall.rs

```diff
@@ -47,6 +51,33 @@ impl CommandExecute for Uninstall {
             ));
         }
 
+        // During install, `harmonic` will store a copy of itself in `/nix/harmonic`
+        // If the user opted to run that particular copy of Harmonic to do this uninstall,
+        // well, we have a problem, since the binary would delete itself.
+        // Instead, detect if we're in that location, if so, move the binary and `execv` it.
+        if let Ok(current_exe) = std::env::current_exe() {
+            if current_exe.as_path() == Path::new("/nix/harmonic") {
+                tracing::debug!(
+                    "Detected uninstall from `/nix/harmonic`, moving executable and re-executing"
+                );
+                let temp = std::env::temp_dir();
+                let temp_exe = temp.join("harmonic");
```

Comment:
The Rust docs have this to say about using `temp_dir` with a fixed name:

> Creating a file or directory with a fixed or predictable name may result in “insecure temporary file” security vulnerabilities. Consider using a crate that securely creates temporary files or directories.

Should we worry about this?

Reply:
Hmmm I think the risk would be if a normal user somehow replaced `/tmp/harmonic` (or whatever) with a malicious binary between us copying it there and running it?

I think that's fair, we can drop some gross random value at the end to make this a lot harder.

Good catch! :)